src/gradle-it/build.gradle

```diff
@@ -0,0 +1,60 @@
+apply plugin: 'java'
+apply plugin: 'maven'
+
+compileJava {
+    targetCompatibility = "1.8"
+    sourceCompatibility = "1.8"
+    options.encoding = 'UTF-8'
+}
+
+description = "Example project that shows how to use gradle to build and deploy artifacts to Strongbox"
+
+ext.username = System.getProperty('credentials.username') != null ? System.getProperty('credentials.username') : mavenUser
+ext.password = System.getProperty('credentials.password') != null ? System.getProperty('credentials.password') : mavenPassword
+
+configurations {
+    deployerJars
+}
+
+repositories {
```

Comment:
Here you need just one repository:
```
maven { url "http://localhost:48080/storages/public/public-group/" }
```

Basically, you should deploy artifacts to the hosted repositories and resolve them via the group. 

Quick rule of thumb:
* Hosted repositories (literally) host artifacts. You can deploy to them.
* Proxy repositories can only cache requested artifacts after resolving them from a remote. You cannot deploy to them.
* Group repositories are used to serve a list of repositories under the same URL. Artifacts are resolved based on the order in which repositories are listed in the group. You cannot deploy to group repositories.

Please, drop these:
```
    maven { url "http://localhost:48080/storages/storage0/releases/" }
    maven { url "http://localhost:48080/storages/storage0/snapshots/" }
```